messaging/src/main/java/org/axonframework/eventhandling/deadletter/jdbc/GenericDeadLetterTableFactory.java

```diff
@@ -0,0 +1,110 @@
+
+    @SuppressWarnings("SqlNoDataSourceInspection")
+    @Override
+    public PreparedStatement createTable(Connection connection, DeadLetterSchema schema) throws SQLException {
+        String sql = "CREATE TABLE IF NOT EXISTS " + schema.deadLetterTable() + " (\n" +
+                schema.deadLetterIdentifierColumn() + " VARCHAR(255) NOT NULL,\n" +
+                schema.processingGroupColumn() + " VARCHAR(255) NOT NULL,\n" +
+                schema.sequenceIdentifierColumn() + " VARCHAR(255) NOT NULL,\n" +
+                schema.sequenceIndexColumn() + " BIGINT NOT NULL,\n" +
+                schema.messageTypeColumn() + " VARCHAR(255) NOT NULL,\n" +
+                schema.eventIdentifierColumn() + " VARCHAR(255) NOT NULL,\n" +
+                schema.timestampColumn() + " " + timestampType() + " NOT NULL,\n" +
+                schema.payloadTypeColumn() + " VARCHAR(255) NOT NULL,\n" +
+                schema.payloadRevisionColumn() + " VARCHAR(255),\n" +
+                schema.payloadColumn() + " " + serializedDataType() + " NOT NULL,\n" +
+                schema.metaDataColumn() + " " + serializedDataType() + ",\n" +
+                schema.aggregateTypeColumn() + " VARCHAR(255),\n" +
+                schema.aggregateIdentifierColumn() + " VARCHAR(255),\n" +
+                schema.sequenceNumberColumn() + " BIGINT,\n" +
+                schema.tokenTypeColumn() + " VARCHAR(255),\n" +
+                schema.tokenColumn() + " " + serializedDataType() + ",\n" +
+                schema.enqueuedAtColumn() + " " + timestampType() + " NOT NULL,\n" +
+                schema.lastTouchedColumn() + " " + timestampType() + ",\n" +
+                schema.processingStartedColumn() + " " + timestampType() + ",\n" +
+                schema.causeTypeColumn() + " VARCHAR(255),\n" +
+                schema.causeMessageColumn() + " VARCHAR(1023),\n" +
+                schema.diagnosticsColumn() + " " + serializedDataType() + ",\n" +
+                "CONSTRAINT PK PRIMARY KEY (" + schema.deadLetterIdentifierColumn() + "),\n" +
+                "CONSTRAINT " + schema.sequenceIndexColumn() + "_INDEX UNIQUE (" +
+                schema.processingGroupColumn() + "," +
+                schema.sequenceIdentifierColumn() + "," +
+                schema.sequenceIndexColumn() +
+                ")\n)";
+        return connection.prepareStatement(sql);
+    }
+
+    @Override
+    public PreparedStatement createProcessingGroupIndex(Connection connection,
+                                                        DeadLetterSchema schema) throws SQLException {
+        String sql = "CREATE INDEX " + schema.processingGroupColumn() + "_INDEX "
+                + "ON " + schema.deadLetterTable() + " "
+                + "(" + schema.processingGroupColumn() + ")";
+        return connection.prepareStatement(sql);
+    }
+
+    @Override
+    public PreparedStatement createSequenceIdentifierIndex(Connection connection,
```

Comment:
This is not a unique index and might lead to problems in case of token stealing

Reply:
Well, the version in the `DeadLetterEntry` isn't unique either :thinking: 
Is that a bug perhaps?

Reply:
Nope, I saw it wrong, it's okay and I'll resolve the comment